### PR TITLE
Adding a language switch to mobile

### PR DESCRIPTION
**Pull request by Angamanga**
This PR adds a translation-UI to the mobile-app, maps strings to translation-files and some translations to Spanish.

Note: The UI is a bit different than the designs in the ticket, this was done in conversation with Reuben.

To test:
1. Go to mobile app
2. Go to profile
3. Tap "Language"
4. Select Spanish
- [x] The radio-button is filled and the text "Current" moves to Spanish
Go out of Language and click on "Log in or Sign up"
- [x] Note that some of the UI is translated

Note 1: Not all strings are translated yet. 
Note 2: A lot of files are changed in this PR! I have noted in the code which ones to focus on, the rest is "just" the language-strings.

### CODE REVIEW

File: apps/mobile-mzima-client/src/app/profile/language/select-language.module.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from '@angular/core';
```
Comment (Angamanga) [inline apps/mobile-mzima-client/src/app/profile/language/select-language.module.ts:1]: The select-language- files are new logic to review.

File: apps/mobile-mzima-client/src/theme/list.scss [hunk 2]
```diff
@@ -63,3 +63,105 @@
     justify-content: center;
   }
 }
+.search-list {
```
Comment (Angamanga) [inline apps/mobile-mzima-client/src/theme/list.scss:66]: Some refactoring to make the list-type usable in more places

File: libs/sdk/src/lib/models/language.interface.ts
Comment (Angamanga) [inline libs/sdk/src/lib/models/language.interface.ts:1]: Moved the interface to the sdk

File: apps/mobile-mzima-client/src/app/shared/components/deployment-item/deployment-item.component.scss [hunk 3]
```diff
@@ -1,82 +1,7 @@
-.deployment-item {
```
Comment (Angamanga) [inline apps/mobile-mzima-client/src/app/shared/components/deployment-item/deployment-item.component.scss:1]: Moved styles to list.scss

File: apps/mobile-mzima-client/src/app/core/constants/profile-menu.ts [hunk 4]
```diff
@@ -67,6 +67,11 @@ export const profileMenu: ProfileMenuItem[] = [
 ];
 
 export const profileInformationMenu: ProfileMenuItem[] = [
+  {
+    label: 'Language',
+    icon: 'language',
+    route: '/profile/select-language',
+  },
   {
```
Comment (Angamanga) [inline apps/mobile-mzima-client/src/app/core/constants/profile-menu.ts:75]: Added language to the profile-menu

File: apps/mobile-mzima-client/src/app/app.module.ts [hunk 5]
```diff
@@ -30,6 +30,10 @@ export const loadConfigProvider: FactoryProvider = {
   multi: true,
 };
 
+export function HttpLoaderFactory(http: HttpClient): any {
+  return new CustomTranslateHttpLoader(http, './assets/locales/', '.json');
+}
```
Comment (Angamanga) [inline apps/mobile-mzima-client/src/app/app.module.ts:35]: For ngx-translate to load the translation-files

File: apps/mobile-mzima-client/src/app/core/services/language.service.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
```
Comment (Angamanga) [inline apps/mobile-mzima-client/src/app/core/services/language.service.ts:1]: These files are more or less copied from the web-client. I think we should be able to move it to the sdk once we have optimised the translation-files and removed duplicates etc so we can use the same files for mobile and web.

File: apps/mobile-mzima-client/src/app/profile/language/select-langugue-routing.module.ts
Comment (ushahidlee) [inline apps/mobile-mzima-client/src/app/profile/language/select-langugue-routing.module.ts:1]: The filename has a typo :stuck_out_tongue_winking_eye:

**Review by ushahidlee — APPROVED** [review]
Works well and the code looks fine. Massive PR!

**Pull request merged**